### ISSUE TITLE
IAP not prepared. Check if Google Play service is available

### ISSUE THREAD

**Issue by vytautas-pranskunas-** [dooboolab/flutter_inapp_purchase#289]
### Version of flutter_inapp_purchase
5.0.1

### Platforms you faced the error (IOS or Android or both?)
both

### Expected behavior
No error when calling await FlutterInappPurchase.instance.initConnection

### Actual behavior
I get error randomly. I get this error, re-open the app ant then there is no error. This bahaviour oberved in emulator AND real devices even on prod buid.

**Comment by muhibbin-munna** [comment]
Did you find any solution?

**Comment by vytautas-pranskunas-** [comment]
Nop.. If maitainers not pay attention to it in worse case i will wrap it into try...catch and will retry initialization :(

**Comment by vytautas-pranskunas-** [comment]
So I have implemented retries - does not help.... I noticed this especialy if i have two devices opened side by side logged in to App store with same user on both. But again - it happens completly randomly because if i restart app i have no error even if same 2 devices running.

**Comment by thealteria** [comment]
getting the same error when I'm trying to `getProducts`. Even tho it's in testing phase.

**Comment by murattyilmaz** [comment]
Same issue here. Any thoughts?

**Comment by thealteria** [comment]
Okay so I guess I figured it out. It's giving me the error when I'm trying to call `getAvailablePurchases()` and `getPurchaseHistory()`. Since I'm on a test account (no merchant account set up yet) so I guess it's giving me error on fetching `getAvailablePurchases()` and `getPurchaseHistory()`. 

I commented those two and it's working fine. But now getting error while purchasing test products. 4 days ago it was working fine too.

Also don't forget to run the `await FlutterInappPurchase.instance.initConnection` in `initState` only otherwise it won't work either.
